rest/route/agent.go

```diff
@@ -1546,3 +1544,104 @@ func (h *checkRunHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return gimlet.NewJSONResponse(fmt.Sprintf("Successfully created check run for  '%v'", t.Id))
 }
+
+// GET /rest/v2/task/{task_id}/github_dynamic_access_token/{owner}/{repo}
+type createGitHubDynamicAccessToken struct {
+	owner  string
+	repo   string
+	taskID string
+
+	permissions github.InstallationPermissions
+
+	env evergreen.Environment
+}
```

Comment:
I hate to do this, but I realized we have a `makeCreateInstallationToken` route and theres a lot of overlap. Instead of a new route, can we edit that route to accommodate this use case?

Reply:
Let me look in to it!

Reply:
Okay, looking at the existing one- I think we could combine it but I'm not sure if we should.

They do have calling "CreateInstallationToken" similar but that one doesn't do any processing with the task, project,  the project's token, and intersection of permissions.

I think the routes also have a different purpose/intent. The existing one is to get our app and it's used to clone projects/modules. While this one gets the user's app from the project the task is on, then intersects it with the given permissions.

I'm not sure what's our end-goal, but I think combining them is a premature optimization- would you be okay with a follow up ticket to consolidate them?

Reply:
I think that if we want to eventually consolidate them we should do it now. 

However, I hear the argument that this adds enough logic to warrant a separate route - I'm okay with that. I think we should maybe add some more comments to explain the difference between the two routes so that people reading it or making edits to it in the future have access to the context.

Reply:
Okay I'll comment both of them to explain what they are used for / why!